### ISSUE TITLE
NVDA is announcing 'document' when dialog pops up.

### ISSUE THREAD

**Issue by VivianSi233** [microsoft/fluentui#22042]
### Environment Information

- Fluent UI React 8.60.1
- **Browser and OS versions**: Edge 99.0.1150.36 OS: Win10

### Describe the issue:

  * Go to https://developer.microsoft.com/en-us/fluentui#/controls/web/dialog
  * Turn on NVDA
  * Navigate to 'Open dialog' button under default dialog and press enter key to activate it
  * Check actual result

#### Actual behavior:

NVDA is announcing 'document' when dialog pops up.
![image](https://user-images.githubusercontent.com/89379019/157580516-2b00880c-133c-4f32-a857-d894d91bf1ce.png)

Bug is not repro on Narrator.

#### Expected behavior:

Screen reader should not announce 'document'.

**Comment by smhigley** [comment]
VoiceOver requires a nested `role="document"` inside dialogs, or VO's quick nav may not be able to access nodes within the dialog.

A side effect of this is NVDA announcing "document" when entering the dialog.
